lib/ui/platform_dispatcher.dart

```diff
@@ -528,12 +528,38 @@ class PlatformDispatcher {
     }
   }
 
-  String? _sendPlatformMessage(String name,PlatformMessageResponseCallback? callback, ByteData? data) =>
+  String? _sendPlatformMessage(String name, PlatformMessageResponseCallback? callback, ByteData? data) =>
       __sendPlatformMessage(name, callback, data);
 
   @FfiNative<Handle Function(Handle, Handle, Handle)>('PlatformConfigurationNativeApi::SendPlatformMessage')
   external static String? __sendPlatformMessage(String name, PlatformMessageResponseCallback? callback, ByteData? data);
 
+  void sendPortPlatformMessage(
+    String name,
+    ByteData? data,
+    int identifier,
+    SendPort port) {
+    final String? error =
+        _sendPortPlatformMessage(name, identifier, port.nativePort, data);
+    if (error != null) {
+      throw Exception(error);
+    }
+  }
+
+  String? _sendPortPlatformMessage(String name, int identifier, int port, ByteData? data) =>
+      __sendPortPlatformMessage(name, identifier, port, data);
+
+  @FfiNative<Handle Function(Handle, Handle, Handle, Handle)>('PlatformConfigurationNativeApi::SendPortPlatformMessage')
+  external static String? __sendPortPlatformMessage(String name, int identifier, int port, ByteData? data);
+
+  int registerRootIsolate() => __registerRootIsolate();
+  @FfiNative<Int64 Function()>('PlatformConfigurationNativeApi::RegisterRootIsolate')
+  external static int __registerRootIsolate();
+
+  void registerBackgroundIsolate(int rootIsolateId) => __registerBackgroundIsolate(rootIsolateId);
+  @FfiNative<Void Function(Int64)>('PlatformConfigurationNativeApi::RegisterBackgroundIsolate')
+  external static void __registerBackgroundIsolate(int rootIsolateId);
```

Comment:
re our conversation the other day -- what i had in mind here was something like this:

```dart
class RootIsolateToken {
  RootIsolateToken._(this._id);
  final int _id;
}

// ...

  bool get isRootIsolate => __isRootIsolate();
￼  @FfiNative<Bool Function()>('PlatformConfigurationNativeApi::IsRootIsolate')
￼  external static bool __isRootIsolate();

  RootIsolateToken registerRootIsolate() => isRootIsolate ? RootIsolateToken._(__registerRootIsolate()) : throw StateError('Current isolate is not the root isolate.');
￼  @FfiNative<Int64 Function()>('PlatformConfigurationNativeApi::RegisterRootIsolate')
￼  external static int __registerRootIsolate();
￼
￼  void registerBackgroundIsolate(RootIsolateToken rootIsolateToken) => isRootIsolate ? : __registerBackgroundIsolate(rootIsolateToken._id);
￼  @FfiNative<Void Function(Int64)>('PlatformConfigurationNativeApi::RegisterBackgroundIsolate')
￼  external static void __registerBackgroundIsolate(int rootIsolateId);
```

(You might be able to make it slightly better by having a singleton instead of minting a new copy each time, dunno if we care though.)